activities/ActivityTemplate/VueJS/js/components/SugarIcon.js

```diff
@@ -0,0 +1,47 @@
+Vue.component('sugar-icon', {
+	data: function () {
+		return {
+			icon: null
+		}
+	},
+	mounted() {
+		var vm = this;
+		requirejs(["sugar-web/graphics/icon"], function (icon) {
+			vm.icon = icon;
+		});
+	},
+	methods: {
+		colorize: function (element, colors, callback) {
+			this.icon.colorize(element, colors, callback);
+		},
+
+		generateIconWithColors: function (path, colors, callback) {
+			requirejs([`text!${path}`], function (icon) {
+				icon = icon.replace(/(stroke)_color\s\"#?\w*\"/, `stroke_color "${colors.stroke}"`);
+				icon = icon.replace(/(fill)_color\s\"#?\w*\"/, `fill_color "${colors.fill}"`);
+				callback("data:image/svg+xml;base64," + btoa(icon));
+			});
+		},
+
+		colorizeIcon: function (element, colors, callback) {
+			var path = this.getBackgroundURL(element);
+			this.generateIconWithColors(path, colors, function (src) {
+				element.style.backgroundImage = `url(${src})`;
+				if (callback) {
+					callback();
+				}
+			});
+		},
+
+		getBackgroundURL: function (elem) {
```

Comment:
Do we need to expose this method?

Reply:
Not really, it is only used by `colorizeIcon()`. Although there is no concept of 'private' methods in Vue templates. We can have a different naming convention for these methods as suggested in the official [Vue Style Guide](https://vuejs.org/v2/style-guide/#Private-property-names-essential). It also shows declaring a private function outside the template object but in our case, that function will then be available everywhere as a global function.